### PR TITLE
Rpi4

### PR DESCRIPTION
**Pull request by jourau**

### CODE REVIEW

File: apps/Arm/vm_cross_connector/overlay_files/init_scripts/cross_vm_module_init [hunk 1]
```diff
@@ -5,4 +5,9 @@
 # SPDX-License-Identifier: BSD-2-Clause
 #
 
-insmod /lib/modules/4.14.87/kernel/drivers/vmm/connection.ko
+echo "HERE"
```
Comment (axel-h) [inline apps/Arm/vm_cross_connector/overlay_files/init_scripts/cross_vm_module_init:8]: Either make this a proper debug/trace message or remove.

File: apps/Arm/vm_cross_connector/rpi4/devices.camkes [hunk 2]
```diff
@@ -0,0 +1,76 @@
+#include <configurations/vm.h>
```
Comment (axel-h) [inline apps/Arm/vm_cross_connector/rpi4/devices.camkes:1]: please add a copyright/license header

File: apps/Arm/vm_minimal/rpi4/devices.camkes [hunk 3]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020, Data61
+ * Commonwealth Scientific and Industrial Research Organisation (CSIRO)
+ * ABN 41 687 119 230.
+ *
+ * This software may be distributed and modified according to the terms of
+ * the BSD 2-Clause license. Note that NO WARRANTY is provided.
+ * See "LICENSE_BSD2.txt" for details.
+ *
+ * @TAG(DATA61_BSD)
```
Comment (axel-h) [inline apps/Arm/vm_minimal/rpi4/devices.camkes:10]: This is deprecated, please use `SPDX-License-Identifier: BSD-2-Clause`

File: apps/Arm/vm_multi/rpi4/devices.camkes [hunk 4]
```diff
@@ -0,0 +1,101 @@
+#include <configurations/vm.h>
```
Comment (axel-h) [inline apps/Arm/vm_multi/rpi4/devices.camkes:1]: please add a header with copyright, license and a brief description

**Comment by kent-mcleod** [comment]
Hi @jourau ,  it looks like this work adds rpi4 as a supported platform to the vm_minimal, vm_multi and vm_crossconnector apps.  Is this work that you would like to see merged upstream or just looking for feedback on?   Are you able to give more information about the status of the completion state and whether there are other PRs to be made on other camkes-vm related repositories?

File: apps/Arm/vm_cross_connector/settings.cmake [hunk 5]
```diff
@@ -4,16 +4,17 @@
...
     message(FATAL_ERROR "PLATFORM: ${PLATFORM} not supported.
          Supported: ${supported}")
 endif()
 set(VmPCISupport ON CACHE BOOL "" FORCE)
 set(LibUSB OFF CACHE BOOL "" FORCE)
 set(VmInitRdFile ON CACHE BOOL "" FORCE)
+set(VmVirtioConsole ON CACHE BOOL "" FORCE)
 if(${PLATFORM} STREQUAL "qemu-arm-virt")
-    # force cpu
+     # force cpu
```
Comment (axel-h) [inline apps/Arm/vm_cross_connector/settings.cmake:17]: space looks like an accident, seems there is no change intended here?

File: apps/Arm/vm_cross_connector/CMakeLists.txt [hunk 6]
```diff
@@ -35,6 +35,12 @@ elseif("${KernelARMPlatform}" STREQUAL "qemu-arm-virt")
     if(SIMULATION)
         GenerateSimulateScript()
     endif()
+elseif("${KernelARMPlatform}" STREQUAL "rpi4")
+    set(cpp_flags "-DKERNELARMPLATFORM_BCM2711")
```
Comment (axel-h) [inline apps/Arm/vm_cross_connector/CMakeLists.txt:39]: prefer `-DKERNELARMPLATFORM_RPI4` here?

**Pull request closed without merging**